Use PROJECT_BINARY_DIR which is the true build dir for cmake,
and respect CXXFLAGS and LDFLAGS.
===================================================================
--- a/tests/hit/HIT.cmake
+++ b/tests/hit/HIT.cmake
@@ -413,7 +413,12 @@ macro(HIT_ADD_FILES _config _dir _label _parent)
                         set_target_properties(${target} PROPERTIES OUTPUT_NAME ${_target_r} RUNTIME_OUTPUT_DIRECTORY "." LINK_DEPENDS "${HIP_LIB_FILES}" PREFIX "" SUFFIX "")
                 else()
                     # message(STATUS "add_custom_target*: target= ${target}  _buildcmd= ${_buildcmd}")
-                    add_custom_target(${target} COMMAND sh -c "${_buildcmd} -L${CMAKE_CURRENT_SOURCE_DIR}/build/lib -isystem ${CMAKE_CURRENT_SOURCE_DIR}/include")
+					if("${_target_type}" STREQUAL "OBJECT")
+						set(EXTRA_BUILD_FLAGS "")
+					else()
+						set(EXTRA_BUILD_FLAGS "${CMAKE_CXX_FLAGS} ${CMAKE_EXE_LINKER_FLAGS}")
+					endif()
+                    add_custom_target(${target} COMMAND sh -c "${_buildcmd} ${EXTRA_BUILD_FLAGS} -L${PROJECT_BINARY_DIR}/lib -isystem ${CMAKE_CURRENT_SOURCE_DIR}/include")
                 endif()
                 add_dependencies(${_parent} ${target})
                 foreach(_dependency ${_depends})
